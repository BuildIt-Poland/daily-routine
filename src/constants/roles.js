// Roles
export const FRONT_END_ROLE = 'frontend';
export const BACK_END_ROLE = 'backend';
export const DEV_OPS_ROLE = 'devops';

// Role actions
export const BRAG_ROLE_ACTION = 'brag';
export const CONFESS_ROLE_ACTION = 'confess';
