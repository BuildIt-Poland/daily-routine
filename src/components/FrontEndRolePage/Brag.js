import React from 'react';

import { CONFIDENT } from '../../constants/characterMoods';
import { Fox } from '../Characters/';

function Brag() {
  return <Fox mood={CONFIDENT} />;
}

export default Brag;
