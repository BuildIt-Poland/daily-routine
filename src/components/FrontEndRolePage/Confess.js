import React from 'react';

import { SAD } from '../../constants/characterMoods';
import { Fox } from '../Characters/';

function Confess() {
  return <Fox mood={SAD} />;
}

export default Confess;
