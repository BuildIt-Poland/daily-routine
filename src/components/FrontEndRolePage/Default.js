import React from 'react';

import { CONFUSED } from '../../constants/characterMoods';
import { Fox } from '../Characters/';

function Default() {
  return <Fox mood={CONFUSED} />;
}

export default Default;
