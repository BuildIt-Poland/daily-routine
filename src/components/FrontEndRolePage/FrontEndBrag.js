import React from 'react';

import { BRAG } from '../../constants/roleActions';
import { Fox } from '../Characters';

function FrontEndBrag() {
  return <Fox pose={BRAG} />;
}

export default FrontEndBrag;
