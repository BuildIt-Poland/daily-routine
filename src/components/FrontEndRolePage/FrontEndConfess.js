import React from 'react';

import { CONFESS } from '../../constants/roleActions';
import { Fox } from '../Characters';

function FrontEndConfess() {
  return <Fox pose={CONFESS} />;
}

export default FrontEndConfess;
