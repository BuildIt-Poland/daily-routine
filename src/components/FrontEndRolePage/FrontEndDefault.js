import React from 'react';

import { DEFAULT } from '../../constants/roleActions';
import { Fox } from '../Characters';

function FrontEndDefault() {
  return <Fox pose={DEFAULT} />;
}

export default FrontEndDefault;
