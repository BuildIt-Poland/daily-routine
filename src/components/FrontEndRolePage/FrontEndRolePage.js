import React from 'react';
import { Route } from 'react-router-dom';

import { BRAG_ROLE_ACTION, CONFESS_ROLE_ACTION, FRONT_END_ROLE } from '../../constants/roles';
import RoleBar from '../RoleBar';
import ActionButtons from '../ActionButtons';
import Section from '../Section';
import Default from './FrontEndDefault';
import Brag from './FrontEndBrag';
import Confess from './FrontEndConfess';

function FrontEndRolePage() {
  return (
    <>
      <RoleBar label="Front End Engineer" />
      <Section>
        <Route exact path={`/${FRONT_END_ROLE}`} component={Default} />
        <Route exact path={`/${FRONT_END_ROLE}/${BRAG_ROLE_ACTION}`} component={Brag} />
        <Route exact path={`/${FRONT_END_ROLE}/${CONFESS_ROLE_ACTION}`} component={Confess} />
        <ActionButtons role={FRONT_END_ROLE} />
      </Section>
    </>
  );
}

export default FrontEndRolePage;
