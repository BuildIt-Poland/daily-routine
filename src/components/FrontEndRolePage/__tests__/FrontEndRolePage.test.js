import React from 'react';
import { create } from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

import { FRONT_END_ROLE } from '../../../constants/roles';
import FrontEndRolePage from '../FrontEndRolePage';

jest.mock('../../Characters', () => ({
  Fox: 'Fox'
}));
jest.mock('../../ActionButtons', () => 'ActionButtons');
jest.mock('../../RoleBar', () => 'RoleBar');
jest.mock('../../QuoteBubble', () => 'QuoteBubble');
jest.mock('../../Page', () => 'Page');
jest.mock('../../Content', () => 'Content');
jest.mock('../../Section', () => 'Section');

describe('COMPONENT - FrontEndRolePage', () => {
  it('renders correctly', () => {
    const component = create(
      <MemoryRouter initialEntries={[`/${FRONT_END_ROLE}`]}>
        <FrontEndRolePage />
      </MemoryRouter>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
});
