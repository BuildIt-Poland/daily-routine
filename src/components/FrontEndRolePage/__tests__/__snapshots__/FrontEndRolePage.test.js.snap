// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT - FrontEndRolePage renders correctly 1`] = `
<Page
  title="Front End Engineer"
>
  <RoleBar
    label="Front End Engineer"
  />
  <Section>
    <RoleContent>
      <QuoteBubble />
      <Fox
        pose="default"
      />
    </RoleContent>
  </Section>
  <ActionButtons
    role="frontend"
  />
</Page>
`;
